### PR TITLE
[lifx] Handle and provide QuantityType for color-temperature-abs channel

### PR DESCRIPTION
**Pull request by ccutrer**
see https://github.com/openhab/openhab-core/pull/3129

It should be noted I don't have LifX, so I have not tested this.

### CODE REVIEW

File: bundles/org.openhab.binding.lifx/README.md [hunk 1]
```diff
@@ -85,19 +85,19 @@ Thing lifx:colorirlight:porch [ host="10.120.130.4", fadetime=0 ]
...
+| Channel Type ID    | Item Type          | Description                                                                                                                                                      | Thing Types                                                                  |
+|--------------------|--------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------|
+| abstemperature     | Number:Temperature | This channel supports adjusting the color temperature in Kelvin.                                                                                                 | colorlight, colorhevlight, colorirlight, colormzlight, tilelight, whitelight |
+| abstemperaturezone | Number:Temperature | This channel supports adjusting the zone color temperature in Kelvin.                                                                                            | colormzlight                                                                 |
```
Comment (lolodomo) [inline bundles/org.openhab.binding.lifx/README.md:91]: Please also fix the example (demo.items) at the end of README.

**Review by lolodomo — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.lifx/src/main/java/org/openhab/binding/lifx/internal/util/LifxMessageUtil.java [hunk 2]
```diff
@@ -117,6 +128,13 @@ public static int percentTypeToKelvin(PercentType temperature, TemperatureRange
                 temperatureRange.getMaximum() - (temperature.floatValue() * (temperatureRange.getRange() / 100)));
     }
 
+    public static int quantityTypeToKelvin(QuantityType temperature, TemperatureRange temperatureRange) {
+        QuantityType<?> asKelvin = temperature.toInvertibleUnit(Units.KELVIN);
+        assert asKelvin != null;
+
+        return asKelvin.toBigDecimal().intValue();
```
Comment (lolodomo) [inline bundles/org.openhab.binding.lifx/src/main/java/org/openhab/binding/lifx/internal/util/LifxMessageUtil.java:135]: @cweitkamp used `convertedCommand.intValue()` in the hue PR/
Is there a difference between `xxx.intValue()` and `xxx.toBigDecimal().intValue()` ?

File: bundles/org.openhab.binding.lifx/src/main/java/org/openhab/binding/lifx/internal/util/LifxMessageUtil.java [hunk 3]
```diff
@@ -102,9 +105,17 @@ public static PercentType kelvinToPercentType(int kelvin, TemperatureRange tempe
...
+    public static int commandToKelvin(Command temperature, TemperatureRange temperatureRange) {
+        if (temperature instanceof PercentType) {
+            return percentTypeToKelvin((PercentType) temperature, temperatureRange);
+        } else if (temperature instanceof QuantityType) {
+            return quantityTypeToKelvin((QuantityType) temperature, temperatureRange);
+        } else if (temperature instanceof DecimalType) {
+            return decimalTypeToKelvin((DecimalType) temperature, temperatureRange);
+        } else {
+            assert false; // we've already done a type check before calling this method
+            return 0;
```
Comment (wborn) [inline bundles/org.openhab.binding.lifx/src/main/java/org/openhab/binding/lifx/internal/util/LifxMessageUtil.java:117]: If it is unexpected throwing an `IllegalStateException` would be better.

File: bundles/org.openhab.binding.lifx/src/main/java/org/openhab/binding/lifx/internal/util/LifxMessageUtil.java (hunk 2 above)
Comment (wborn) [reply bundles/org.openhab.binding.lifx/src/main/java/org/openhab/binding/lifx/internal/util/LifxMessageUtil.java:135]: Please also don't use assertions.

**Comment by wborn** [comment]
> It should be noted I don't have LifX, so I have not tested this.

I have many LIFX lights so I can test it after the review comments have been addressed. :slightly_smiling_face:

File: bundles/org.openhab.binding.lifx/src/main/java/org/openhab/binding/lifx/internal/util/LifxMessageUtil.java (hunk 2 above)
Comment (ccutrer) [reply bundles/org.openhab.binding.lifx/src/main/java/org/openhab/binding/lifx/internal/util/LifxMessageUtil.java:135]: 🤦  I totally looked for QuantityType().intValue(), and didn't see it.

**Comment by ccutrer** [comment]
Question: what's the etiquette on resolving inline review comments? I naturally defer to letting the author agree that it has been fully addressed, but it would be nice to be able to keep track of what I've addressed so far using GitHub itself (especially now that all of a sudden they're popping up in my IDE; dunno what I did that caused that to work!).

File: bundles/org.openhab.binding.lifx/README.md [hunk 4]
```diff
@@ -175,44 +175,44 @@ Thing lifx:whitelight:kitchen [ deviceId="D073D5D4D4D4", fadetime=150 ]
 // Living
 Color Living_Color { channel="lifx:colorlight:living:color" }
 Dimmer Living_Temperature { channel="lifx:colorlight:living:temperature" }
-Number Living_Abs_Temperature { channel="lifx:colorlight:living:abstemperature" }
+Number:Number Living_Abs_Temperature "Living Room Lights Color Temperature [%d K]" { channel="lifx:colorlight:living:abstemperature" }
```
Comment (wborn) [inline bundles/org.openhab.binding.lifx/README.md:178]:
```suggestion
Number:Temperature Living_Abs_Temperature "Living Room Lights Color Temperature [%d K]" { channel="lifx:colorlight:living:abstemperature" }
```

**Comment by wborn** [comment]
>  I naturally defer to letting the author agree that it has been fully addressed

I also do that. It's also common for contributors to mark something as resolved when they did not resolve anything. :upside_down_face:

Replying with "Fixed" to every review comment is also highly annoying because it causes notifications for all PR subscribers which can sometimes go on for hours when someone is reworking code.

Using :+1: as feedback is a good alternative but it is a bit more  involved.

I usually just rework them from top to bottom so I only have to keep track of where I am with addressing the comments.

**Review by lolodomo — APPROVED** [review]
LGTM, thank you

**Comment by lolodomo** [comment]
> I have many LIFX lights so I can test it after the review comments have been addressed. 🙂

@wborn : would you like to test before I merge ?

**Comment by wborn** [comment]
Yes I'll give it a test soon.

**Review by wborn — APPROVED** [review]
It works fine after doing some tests. :+1:

**Pull request merged**